File: src/HMS.Domain/Entities/Doctor.cs
using HMS.Domain.Common;

namespace HMS.Domain.Entities;

public class Doctor : BaseEntity
{
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;

    public string Specialization { get; set; } = string.Empty;
    public string Qualification { get; set; } = string.Empty;
    public string? LicenseNumber { get; set; }
    public decimal ConsultationFee { get; set; }
    public int ExperienceYears { get; set; }

    // Navigation properties
    public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    public ICollection<Prescription> Prescriptions { get; set; } = new List<Prescription>();
}

File: frontend/src/types/doctor.ts
export interface DoctorDto {
  id: string;
  userId: string;
  fullName: string;
  email: string;
  phoneNumber: string | null;
  specialization: string;
  qualification: string;
  licenseNumber: string | null;
  consultationFee: number;
  experienceYears: number;
  createdAt: string;
}

export interface CreateDoctorDto {
  userId: string;
  specialization: string;
  qualification: string;
  licenseNumber?: string;
  consultationFee: number;
  experienceYears: number;
}

export interface UpdateDoctorDto {
  specialization: string;
  qualification: string;
  licenseNumber?: string;
  consultationFee: number;
  experienceYears: number;
}
